backends/guest/common/verify.c

```diff
@@ -95,13 +96,15 @@ static int get_current_esl_data(const char *esl_file, uint8_t **current_esl_data
 		return INVALID_FILE;
 
 	buffer = get_data_from_file(esl_file, SIZE_MAX, &buffer_size);
-	if (buffer != NULL) {
+	if (buffer != NULL && buffer_size >= TIMESTAMP_LEN) {
 		if (buffer_size == DEFAULT_PK_LEN) {
 			print_raw(buffer, buffer_size);
 			free(buffer);
 			buffer = NULL;
 			buffer_size = 0;
-		} else if (buffer_size != TIMESTAMP_LEN) {
+		} else {
+			if (verbose >= PR_INFO)
+				read_timestamp(buffer);
 			rc = validate_esl(buffer + TIMESTAMP_LEN, buffer_size - TIMESTAMP_LEN);
```

Comment:
Seems our `TIMESTAMP_LEN` macro has a misleading name. Correct me if I am wrong but it is the len of the version + the timestamp?

Also `TIMESTAMP_LEN` is assigned the value of 8 (in `backends/guest/include/common/util.h`) but the definition of `timestamp_t` (in `external/libstb-secvar/external/edk2/common.h`) has  a length of 16?

Reply:
You are correct, and it seems that `TIMESTAMP_LEN` is the length of how much of the timestamp that we actually use. I really don't like this, and I'll have to scan the code in another PR to see if we ever need to parse in the full 16 bytes (e.g. inscribed in an auth header, probably?). I'd rather use different types for parsing different data...

For now, I'm going to add comment to the definition of this constant, where its used, and `read_timestamp()`. I will lump the work to improve this into #76 .


Documenting here just in case it never gets documented properly:
```c
struct sv_timestamp
{
  leint16_t year;
  uint8_t month;
  uint8_t day;
  uint8_t hour;
  uint8_t minute;
  uint8_t second;  // <-- this is how much we actually read in with read_timestamp()
  uint8_t pad1;    // <-- TIMESTAMP_LEN stops here
  leint32_t nanosecond;  // \
  leint16_t timezone;    //  |-- unused section of the struct
  uint8_t daylight;      //  |
  uint8_t pad2;          // /
} SV_PACKED;
```

Reply:
> For now, I'm going to add comment to the definition of this constant, where its used, and read_timestamp(). I will lump the work to improve this into https://github.com/open-power/secvarctl/issues/76 .

JK new commit uses structs because it was easier to implement something better than document it.